File: backend/src/main/java/com/fatec/tarde/loja/Cliente.java
package com.fatec.tarde.loja;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;

@Entity
public class Cliente {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO) 
    private int codigo;
    private String nome;
    private String email;
    private String documento;
    private String logradouro;
    private String senha;
    private int autoriza;
    private String telefone;

    public int getCodigo() {
        return codigo;
    }
    public void setCodigo(int codigo) {
        this.codigo = codigo;
    }
    public String getNome() {
        return nome;
    }
    public void setNome(String nome) {
        this.nome = nome;
    }
    public String getEmail() {
        return email;
    }
    public void setEmail(String email) {
        this.email = email;
    }
    public String getDocumento() {
        return documento;
    }
    public void setDocumento(String documento) {
        this.documento = documento;
    }
    public String getLogradouro() {
        return logradouro;
    }
    public void setLogradouro(String logradouro) {
        this.logradouro = logradouro;
    }
    public String getSenha() {
        return senha;
    }
    public void setSenha(String senha) {
        this.senha = senha;
    }
    public int getAutoriza() {
        return autoriza;
    }
    public void setAutoriza(int autoriza) {
        this.autoriza = autoriza;
    }
    public String getTelefone() {
        return telefone;
    }
    public void setTelefone(String telefone) {
        this.telefone = telefone;
    }
    
}

File: frontend/src/app/model/cliente.ts
export class Cliente {
    codigo: number = 0;
    nome: string = "";
    email: string = "";
    documento: string = "";
    logradouro: string = "";
    senha: string = "";
    autoriza: number = 0; // 0 - Não autoriza, 1 - Autoriza
    telefone: string = "";
}

/**
 * @Id
    @GeneratedValue(strategy = GenerationType.AUTO) 
    private int codigo;
    private String nome;
    private String email;
    private String documento;
    private String logradouro;
    private String senha;
    private String autoriza;
    private String telefone;
 */